caddyhttp/proxy/reverseproxy.go

```diff
@@ -318,30 +318,61 @@ func (rp *ReverseProxy) ServeHTTP(rw http.ResponseWriter, outreq *http.Request,
 		}
 		pooledIoCopy(backendConn, conn)
 	} else {
+		// NOTE:
+		//   Closing the Body involves acquiring a mutex, which is a
```

Comment:
Why would this require a mutex? I'm trying to see where it would be closed in a different goroutine.